### ISSUE TITLE
Duplicate Event Attributes

### ISSUE THREAD

**Issue by dpederson-figure** [CosmWasm/wasmd#445]
## Problem

When submitting a transaction that is executed by either the `init` or `handle` function in a smart contract, when events are emitted, attributes are duplicated. This happens for all event types except `message`. This only happens during smart contract transactions. No dup events are produced when calling blockchain modules directly via RPC or CLI.

wasmd version =  v0.15.1 ; See [here](https://github.com/provenance-io/provenance/blob/v0.2.0/go.mod#L6)

## Example

See [here](https://github.com/provenance-io/provwasm/blob/main/contracts/name/README.md) as an example of the steps to demonstrate this issue.

Simply storing then instantiating the above integration test contract will produce the following output.

```json
{
  "height": "17",
  "txhash": "9B84C4DE0C8582DE5D6B10AC227EECA57C0ACA1EDB9F6DC9BE80C44C8658039F",
  "codespace": "",
  "code": 0,
  "data": "0A3A0A0B696E7374616E7469617465122B0A2974703138766438667077787A636B3933716C776768616A3661726834703763356E383978386B736B7A",
  "raw_log": "[{\"events\":[{\"type\":\"message\",\"attributes\":[{\"key\":\"action\",\"value\":\"instantiate\"},{\"key\":\"module\",\"value\":\"wasm\"},{\"key\":\"signer\",\"value\":\"tp1f2ecech5swn8na2scfdxshhqjpapgdcq9dywh0\"},{\"key\":\"code_id\",\"value\":\"1\"},{\"key\":\"contract_address\",\"value\":\"tp18vd8fpwxzck93qlwghaj6arh4p7c5n89x8kskz\"}]},{\"type\":\"name_bound\",\"attributes\":[{\"key\":\"address\",\"value\":\"tp18vd8fpwxzck93qlwghaj6arh4p7c5n89x8kskz\"},{\"key\":\"name\",\"value\":\"name-itv2.sc.pb\"},{\"key\":\"address\",\"value\":\"tp18vd8fpwxzck93qlwghaj6arh4p7c5n89x8kskz\"},{\"key\":\"name\",\"value\":\"name-itv2.sc.pb\"}]},{\"type\":\"wasm\",\"attributes\":[{\"key\":\"contract_address\",\"value\":\"tp18vd8fpwxzck93qlwghaj6arh4p7c5n89x8kskz\"},{\"key\":\"integration_test\",\"value\":\"v2\"},{\"key\":\"action\",\"value\":\"provwasm.contracts.name.init\"},{\"key\":\"contract_name\",\"value\":\"name-itv2.sc.pb\"},{\"key\":\"contract_owner\",\"value\":\"tp1f2ecech5swn8na2scfdxshhqjpapgdcq9dywh0\"},{\"key\":\"contract_address\",\"value\":\"tp18vd8fpwxzck93qlwghaj6arh4p7c5n89x8kskz\"},{\"key\":\"integration_test\",\"value\":\"v2\"},{\"key\":\"action\",\"value\":\"provwasm.contracts.name.init\"},{\"key\":\"contract_name\",\"value\":\"name-itv2.sc.pb\"},{\"key\":\"contract_owner\",\"value\":\"tp1f2ecech5swn8na2scfdxshhqjpapgdcq9dywh0\"}]}]}]",
  "logs": [
    {
      "msg_index": 0,
      "log": "",
      "events": [
        {
          "type": "message",
          "attributes": [
            {
              "key": "action",
              "value": "instantiate"
            },
            {
              "key": "module",
              "value": "wasm"
            },
            {
              "key": "signer",
              "value": "tp1f2ecech5swn8na2scfdxshhqjpapgdcq9dywh0"
            },
            {
              "key": "code_id",
              "value": "1"
            },
            {
              "key": "contract_address",
              "value": "tp18vd8fpwxzck93qlwghaj6arh4p7c5n89x8kskz"
            }
          ]
        },
        {
          "type": "name_bound",
          "attributes": [
            {
              "key": "address",
              "value": "tp18vd8fpwxzck93qlwghaj6arh4p7c5n89x8kskz"
            },
            {
              "key": "name",
              "value": "name-itv2.sc.pb"
            },
            {
              "key": "address",
              "value": "tp18vd8fpwxzck93qlwghaj6arh4p7c5n89x8kskz"
            },
            {
              "key": "name",
              "value": "name-itv2.sc.pb"
            }
          ]
        },
        {
          "type": "wasm",
          "attributes": [
            {
              "key": "contract_address",
              "value": "tp18vd8fpwxzck93qlwghaj6arh4p7c5n89x8kskz"
            },
            {
              "key": "integration_test",
              "value": "v2"
            },
            {
              "key": "action",
              "value": "provwasm.contracts.name.init"
            },
            {
              "key": "contract_name",
              "value": "name-itv2.sc.pb"
            },
            {
              "key": "contract_owner",
              "value": "tp1f2ecech5swn8na2scfdxshhqjpapgdcq9dywh0"
            },
            {
              "key": "contract_address",
              "value": "tp18vd8fpwxzck93qlwghaj6arh4p7c5n89x8kskz"
            },
            {
              "key": "integration_test",
              "value": "v2"
            },
            {
              "key": "action",
              "value": "provwasm.contracts.name.init"
            },
            {
              "key": "contract_name",
              "value": "name-itv2.sc.pb"
            },
            {
              "key": "contract_owner",
              "value": "tp1f2ecech5swn8na2scfdxshhqjpapgdcq9dywh0"
            }
          ]
        }
      ]
    }
  ],
  "info": "",
  "gas_wanted": "149128",
  "gas_used": "147619",
  "tx": null,
  "timestamp": ""
}
```

It can be seen that the attributes have been duplicated for both our custom blockchain module (type `name_bound`), as well as those attributes emitted by the integration test smart contract (type `wasm`).

**Comment by dpederson-figure** [comment]
It seems that with the release of wasmd v0.16.0, we're no longer seeing dup events in Provenance for our smart contracts.  Unsure if this is related to our updates or wasmd updates, but it seems this is no longer an issue.  Feel free to close unless you'd like to investigate further.

**Comment by ethanfrey** [comment]
Yeah, there were a lot of fixes in wasmd 0.16. I assume this one was intentionally fixed, just not closed on the issue.

@alpe you meant to do this, right? Want to close?

**Comment by alpe** [comment]
This still needs to be fixed but it did not make it into v0.16.0. We also have https://github.com/CosmWasm/wasmd/issues/448 which seems to be related

**Comment by alpe** [comment]
I have watched some events on init and can not see or replicate the issue.
Closing this now.
